modules/gdscript/gdscript_compiler.cpp

```diff
@@ -2277,14 +2277,13 @@ Error GDScriptCompiler::_populate_class_members(GDScript *p_script, const GDScri
 					_set_error(vformat(R"(Could not compile base class "%s" from "%s": %s)", base->fully_qualified_name, base->path, error_names[err]), nullptr);
 					return err;
 				}
-				if (base_root.is_valid()) {
+				if (!base_root.is_null()) {
 					base = Ref<GDScript>(base_root->find_class(base->fully_qualified_name));
 				}
 				if (base.is_null()) {
 					_set_error(vformat(R"(Could not find class "%s" in "%s".)", base->fully_qualified_name, base->path), nullptr);
 					return ERR_COMPILATION_FAILED;
 				}
-				ERR_FAIL_COND_V(!base->is_valid(), ERR_BUG);
```

Comment:
this shouldn't be removed, if the base isn't valid here, theres something wrong elsewhere,
either `get_full_script()` should have given an error, or the inner class is not valid for some reason
the code i replaced didn't allow this

Reply:
In the case of the MRP of #69213, `base` is reloading. While reloading, the script `valid` value is set to false, as shown:

https://github.com/godotengine/godot/blob/860884bde7d895887236306887d01381f747bcd7/modules/gdscript/gdscript.cpp#L850-L860

So, as replied to @kleonc, in `gdscript.h`, `GDScript::is_valid()` returns a different value than being the inverse of `GDScript::is_null()`. An invalid script just means that it's reloading. We could change the behaviour, but checking if the script is valid, at this stage, just creates bug, it doesn't prevent ones.

https://github.com/godotengine/godot/blob/860884bde7d895887236306887d01381f747bcd7/modules/gdscript/gdscript.h#L185

Maybe we could add a check to `is_valid()` that it must be `GDScript::reloading == true` too. Otherwise, it's a bug.

Reply:
I put back the error with the added `&& !base->reloading` condition.

Reply:
that sounds good to me 👍